build.gradle.kts

```diff
@@ -37,3 +42,67 @@ detekt {
         html.enabled = false // Disabled due to requirement for kotlinx-html which is still in JCenter.
     }
 }
+
+java {
+    withJavadocJar()
+    withSourcesJar()
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            artifactId = "kached-properties"
+            from(components["java"])
+
+            pom {
+                name.set("kached-properties")
+                description.set("Caching in Kotlin made simple.")
+                url.set("https://github.com/dynatrace-oss/kached-properties")
+
+                licenses {
+                    license {
+                        name.set("Apache License, version 2.0")
+                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                    }
+                }
+
+                scm {
+                    connection.set("scm:git:git://github.com/dynatrace-oss/kached-properties.git/")
+                    developerConnection.set("scm:git:ssh://github.com:dynatrace-oss/kached-properties.git")
+                    url.set("https://github.com/dynatrace-oss/kached-properties.git")
+                }
+
+                developers {
+                    developer {
+                        id.set("krzema12")
+                        name.set("Piotr Krzemiński")
+                        email.set("piotr.krzeminski@dynatrace.com")
+                    }
+                    developer {
+                        id.set("urielsalis")
+                        name.set("Uriel Salischiker")
+                        email.set("uriel.salischiker@dynatrace.com")
+                    }
+                }
+            }
+        }
+    }
+
+    val ossrhUsername: String? by project
+    val ossrhPassword: String? by project
+
+    repositories {
+        maven(url = "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
```

Comment:
Why `oss.sonatype.org`, not `s01.oss.sonatype.org`: https://central.sonatype.org/publish/publish-guide/

> Note: As of February 2021, all new projects began being provisioned on https://s01.oss.sonatype.org/. If your project is not provisioned on https://s01.oss.sonatype.org/, you will want to login to the legacy host https://oss.sonatype.org/.